higlass/tilesets.py

```diff
@@ -106,6 +106,19 @@ def beddb(filepath, uuid=None, **kwargs):
     )
 
 
+def bed2ddb(filepath, uuid=None, **kwargs):
```

Comment:
Would you mind if we hold off merging the PR and include an option to support 1D and 2D bedpedb files? I created a not merged yet extension to clodius for bedpedb 1D tiles support. Since bedpe can be rendered either as 1D or 2D I wonder if we could change the function signature to:

```def bedpedb(filepath, uuid=None, serve_1d_tiles=False, **kwargs):```

And then conditionally switch between 2D (default) or 1D tiles.

What do you think?

Reply:
How can bedpe be rendered as 1d tiles?

Reply:
As regions, arcs, or arc-bands. There's already a `get_1d_tiles()` function in clodius. I extended it and use it with 1.5D bed annotations (the start site is a region but the endpoint is just a point)

Reply:
I'm not sure I understand what it would return. Like if you say get me tile 1.0, how do you interpret the y coordinates?

For the horizontal-heatmap track, we display it in  a 1D  track we still use 2d tiles, the coordinate system is  just inverted so that the diagonal is along the x-axis.

Reply:
The return value is the same. Apparently only the indexing differs:

1D query: https://github.com/higlass/clodius/blob/develop/clodius/tiles/bed2ddb.py#L95-L102
2D query: https://github.com/higlass/clodius/blob/develop/clodius/tiles/bed2ddb.py#L148-L156

It appears as if for 1D tiles only the first region needs to be within the tile and for 2D the first and second region needs to be within.

I think I also found a bug: regions that span 2 tiles would never be retrieved.

Reply:
Ho hum. I suppose that's OK then. But there should be a limit on the number of values retrieved. The zoom levels are calculated for 2D tiles and I'm concerned that fetching a 1D tile could return a huge amount of data.